### PR TITLE
Allow to set custom scalyr_server via environment variable

### LINKED ISSUE #22: Allow setting Scalyr server as environment variable.
I am using exclusively https://eu.scalyr.com, and it would be very handy to enable setting `--server` parameter through environment variables in the same way as the API keys are set. E.g.
```
export scalyr_server='https://eu.scalyr.com'
```

### PR DESCRIPTION
**Pull request by femueller**
This PR would fix https://github.com/scalyr/scalyr-tool/issues/22

### CODE REVIEW

**Comment by imron** [comment]
Hi femueller, thanks for this.

Before I merge this, could you make it so that if the command line option is specified that it overrides the environment variable?  This will make things more in line with common unix precedence behavior.

In order to detect if the server parameter has been set on the command line, you'll need to remove the default value in the ArgumentParser, and then check for None.  If this is None *and* the env variable is also not set, then manually set the value to the current default.

**Comment by femueller** [comment]
Hey @imron,

I've adjusted and tested the required changes.

File: scalyr
Comment (imron) [inline scalyr]: Could you remove the default='', so this line just looks like:

    parser.add_argument('--server',

If a default isn't specified, then the default value be None, and I think it's better to use None if no value is specified rather than an empty string.

The reason for this is that if testing against an empty string and someone runs the command from bash:

    ./scalyr --server "$SCALYR_EU_SERVER" .....

But they forget to actually set $SCALYR_EU_SERVER in their shell, then it will evaluate to `./scalyr --server "" ...` which is an empty string, and so it would then silently connect to the `https://www.scalyr.com` address rather than the presumed https://eu.scalyr.com

By contrast, if None is the default, then an empty string for the server will fail out with a connection error.

The logic on lines 93-95 will then also need to be changed to check for None rather than ''.

**Comment by femueller** [comment]
I've now simplified the if condition by setting a default os.environment value and replaced empty string comparision with ``None``.

**Review by imron — APPROVED** [review]
Great.  That all looks good.

Thanks for taking the time to put this together :-)

**Pull request merged**
